### DISCUSSION TITLE
Augmentations in batch-size level

### DISCUSSION THREAD

**Discussion by black0017** [TorchIO-project/torchio#562] [General]
Hello @fepegar , really great work!

One short question: is it possible to apply the implemented augmentations on a batch-level (5D tensor)?

### Quick example
```
import torch
import torchio.transforms as T
x = torch.rand(16, 1, 1, 256, 256)
transf = T.RandomElasticDeformation() # or # transf = T.Crop(cropping=(0, 10, 10)) 
y = transf(x)
``` 
Results in:
```
ValueError: The input must be a 4D tensor with dimensions (channels, x, y, z) but it has shape (16, 1, 1, 256, 256)
```
Why is it not possible to apply transformations on a batch-level? Is there any quick fix to it?

Thanks in advance,
Nikolas

**Comment by fepegar** [comment]
Hi, Nikolas. Thanks for your kind words!

What about just iterating over your batch?

```python
import torch
import torchio.transforms as T
x = torch.rand(16, 1, 1, 256, 256)
transf = T.RandomElasticDeformation()
y = torch.stack([transf(instance) for instance in x])
```

**Reply by black0017** [reply]
Nice, I see. Thanks a lot for the reply @fepegar !!!

**Reply by fepegar** [reply]
No worries!
